src/app/space/create/deployments/deployments-linechart/deployments-linechart.component.ts

```diff
@@ -0,0 +1,178 @@
+import {
+  Component,
+  DoCheck,
+  Input,
+  OnInit,
+  ViewEncapsulation
+} from '@angular/core';
+
+import {
+  ChartBase,
+  ChartDefaults
+} from 'patternfly-ng';
+
+import {
+  cloneDeep,
+  defaultsDeep,
+  isEqual,
+  merge,
+  uniqueId
+} from 'lodash';
+
+import { DeploymentsLinechartConfig } from './deployments-linechart-config';
+import { DeploymentsLinechartData } from './deployments-linechart-data';
+
+@Component({
+  encapsulation: ViewEncapsulation.None,
+  selector: 'deployments-linechart',
+  templateUrl: './deployments-linechart.component.html',
+  styleUrls: ['./deployments-linechart.component.less']
+})
+export class DeploymentsLinechartComponent extends ChartBase implements DoCheck, OnInit {
+
+  @Input() chartData: DeploymentsLinechartData;
+
+  @Input() config: DeploymentsLinechartConfig;
+
+  private defaultConfig: DeploymentsLinechartConfig;
+  private prevChartData: DeploymentsLinechartData;
+  private prevConfig: DeploymentsLinechartConfig;
+
+  constructor(private chartDefaults: ChartDefaults) {
+    super();
+  }
+
+  ngOnInit(): void {
+    this.setupConfigDefaults();
+    this.setupConfig();
+    this.generateChart(this.config, true);
+  }
+
+  ngDoCheck(): void {
+    if (!isEqual(this.config, this.prevConfig) || !isEqual(this.chartData, this.prevChartData)) {
+      this.setupConfig();
+      this.generateChart(this.config, isEqual(this.chartData, this.prevChartData));
```

Comment:
@andrewazores Oh. I just realized; I think when loading data via chart.load, by default, it doesn't remove any of the old data. You can specify to unload all prior to loading or to load only the new data. I'm not sure how easy it is to do in this situation. Can you verify if this is the case?

Reply:
I see. Is the concern that without unloading the old data, the c3 component will render duplicates of each old data point? I see the documentation about the "unload" parameter when calling the load function of a c3 chart, but it's not exactly clear when or why this would be required.

Reply:
Yes. I ran into this with the c3js donut chart and though I haven't tested it, I think the other c3js charts follow the same behavior. The load's default behavior is more of an 'append'. For the donut chart, it was okay for me to just set 'config.data.unload' to true to unload all the previous data before adding the replacement data. It's not that expensive; I suppose if these charts are limited to 15 data points it wouldn't be that expensive either.

Reply:
I manually tested to check for this behaviour with this patch:

https://paste.fedoraproject.org/paste/A5sAGbyfgV-xkQBq-xkqOw

Examining the console output, I don't see any difference in the behaviour of the internal data state of the c3 component with or without the "unload" property set. Either way, the data set grows by a single data point per update as expected, with no duplicates in the array or apparent changes to any prior data points. Perhaps this is just not the right internal state to examine.

I can include this property if you insist, but I'm not entirely sure of what exact effect it has on this component.

Reply:
@andrewazores It looks like it's fine. I did notice another issue where

```
isEqual(this.chartData, this.prevChartData)
```

is always true because there is a clone deep in the `setupConfig` call prior so it always regenerates anyways. The result prior to the `setupConfig` call needs to be extracted to a boolean.

Reply:
Ah thanks, nice catch.

Reply:
I re-tested the `unload` property with that change and the behaviour still looks correct with or without the property. The chart visibly behaves more nicely with the constant extracted since that allows us to actually avoid the regeneration as intended, so there's no visual flicker on update - just a smooth update of the chart dataset.

Perhaps we should make a PR to upstream pf-ng to enhance the sparkline component similarly?

Reply:
Yeah my tests had a similar result. It doesn't reload the dataset and only adds what's new.